### ISSUE TITLE
Extract autoloads from packages after installation

### ISSUE THREAD

**Issue by makeamark1** [Homebrew/homebrew-emacs#752]
Emacs has the concept of [autoloading](https://www.gnu.org/software/emacs/manual/html_node/elisp/Autoload.html), which allows registering functions but deferring loading of them until a later time to reduce startup time. This is especially useful if a user has many packages installed. It also removes the need for a user to add explicit loading of the package to the user's init file. Most Emacs packages already provide autoload cookies and the built-in Emacs package manager extracts them after installation, which convenient for the user.

Could the Homebrew Emacs tap be made support extracting autoloads too? It would probably involve executing an Emacs function after package installation to extract the autoload cookies into `/usr/local/share/emacs/site-lisp`. It might be good to look at how the Emacs built-in package manager does it for reference. If support for autoloads is likely to be an accepted feature of the Homebrew Emacs tap, I could look into contributing through a pull request.

**Comment by dunn** [comment]
Yes, that would be very welcome!  Another user began work on it a while back, but never quite finished: https://github.com/dunn/homebrew-emacs/pull/13
